src/main/java/me/karakelley/http/controllers/StaticFilesController.java

```diff
@@ -4,27 +4,55 @@
 import me.karakelley.http.Request;
 import me.karakelley.http.Response;
 import me.karakelley.http.Status;
+import me.karakelley.http.utility.HtmlGenerator;
 
+import java.io.File;
 import java.util.ArrayList;
 import java.util.Collections;
+import java.util.List;
 
 public class StaticFilesController implements Controller {
-  private final ArrayList availableActions = new ArrayList<>(Collections.singleton("GET"));
+
+  public final ArrayList availableActions = new ArrayList<>(Collections.singleton("GET"));
+  private final HtmlGenerator htmlGenerator;
+  private final PublicDirectory publicDirectory;
+
+  public StaticFilesController(PublicDirectory publicDirectory, HtmlGenerator htmlGenerator) {
```

Comment:
Why do you need to inject an instance of `HtmlGenerator`?

Reply:
I wanted to generate html when I set the response body. Before, I was using a static method to generate the HTML but it was tightly coupled so I decided to use dependency injection.

Reply:
I would argue this is still pretty tightly coupled (although less so than using a `static` method). You could change this implementation if you sub-classed `HtmlGenerator` and overrode all its methods. That would be pretty weird, though, so it's probably better to depend on an `interface` or `abstract class` to reduce coupling.

But, lets pretend `HtmlGenerator` *is* an `interface`. That naming alone is causing some coupling. What if I wanted to return the directory contents as JSON? What about CSV? XML? Do you really need an `HtmlGenerator` or is there something even more abstract you could depend on?